### ISSUE TITLE
Set code naming standard

### ISSUE THREAD

**Issue by Dillonzer** [Pokemon-Standards-Consortium/standards#8]
Regarding the Set Code (SWSH4, TK10A, etc) are there any limitations we should be thinking of when trying to figure out what to name the missing ones?

-  Should we allow only alphanumeric? 
- Should we also allow special characters? 
- Should there be a length limit?
- Is there some regex it should be passing to be a valid set code?

To make sure we have a standard we should make sure we have a standard convention for naming them as well.

**Comment by immewnity** [comment]
For what it's worth, TCGO uses underscores in their "codes" (e.g. `Promo_XY`) and hyphens in their "abbreviations" (e.g. `PR-XY`). Of course, most places seem to use `XYP` as the code in this case... I think we might be able to fully avoid anything non-alphanumeric.

For lengths, I'm not sure any strict limit would need to be set. What would you see as the primary benefit of setting a limit? Field sizes?

**Comment by Dillonzer** [comment]
I can't think of a useful reason for a limit other than specialty sets getting out of control. (ie #1 #6).

**Comment by egonolieux** [comment]
Are we planning to extend the scope to the Japanese TCG as well in the future? Or are we primarily concerned about PTCGi?

**Comment by immewnity** [comment]
PSC 001 is currently scoped to:
> TCG releases by Wizards of the Coast, Nintendo, Pokémon USA, and/or The Pokémon Company International

Due to how different Japanese releases are structured, maybe a separate document should be made for it? Or we could put it in a subsection of PSC 001, something like "PSC 001-J"?

On a related note, I'm not entirely sure how the other TPC regions would play in (Korea, Indonesia, Thailand, etc.)...

**Comment by egonolieux** [comment]
I don't think we should write a standard for all regions right away, but deciding on a document structure/standard name that supports different regions might be a good thing to decide on from the beginning.

I think suffixing the region like `PSC 001-J` (or `PSC 001-JP`) is a good idea. Maybe for consistency we should also use the `-INTL` or `-PTCGI` suffix? For standards where region does not apply, it should be left out of course.

**Comment by immewnity** [comment]
> 
> 
> I don't think we should write a standard for all regions right away, but deciding on a document structure/standard name that supports different regions might be a good thing to decide on from the beginning.
> 
> I think suffixing the region like `PSC 001-J` (or `PSC 001-JP`) is a good idea. Maybe for consistency we should also use the `-INTL` or `-PTCGI` suffix? For standards where region does not apply, it should be left out of course.

That could work - using ISO 3166-1 alpha-2 codes for the countries, maybe? And then yeah, `-INTL` or similar for the rest.
